pkg/shares/message_shares_test.go

```diff
@@ -0,0 +1,99 @@
+package shares
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/pkg/consts"
+	coretypes "github.com/tendermint/tendermint/types"
+)
+
+func Test_parseMsgShares(t *testing.T) {
+	// exactMsgShareSize is the length of message that will fit exactly into a single
+	// share, accounting for namespace id and the length delimiter prepended to
+	// each message
+	const exactMsgShareSize = consts.MsgShareSize - 2
+
+	type test struct {
+		name     string
+		msgSize  int
+		msgCount int
+	}
+
+	// each test is ran twice, once using msgSize as an exact size, and again
+	// using it as a cap for randomly sized leaves
+	tests := []test{
+		{"single small msg", 100, 1},
+		{"many small msgs", 100, 10},
+		{"single big msg", 1000, 1},
+		{"many big msgs", 1000, 10},
+		{"single exact size msg", exactMsgShareSize, 1},
+		{"many exact size msgs", consts.MsgShareSize, 10},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		// run the tests with identically sized messagses
+		t.Run(fmt.Sprintf("%s idendically sized ", tc.name), func(t *testing.T) {
+			rawmsgs := make([]coretypes.Message, tc.msgCount)
+			for i := 0; i < tc.msgCount; i++ {
+				rawmsgs[i] = generateRandomMessage(tc.msgSize)
+			}
+
+			msgs := coretypes.Messages{MessagesList: rawmsgs}
+			msgs.SortMessages()
+
+			shares, _ := SplitMessages(nil, msgs.MessagesList)
+
+			parsedMsgs, err := parseMsgShares(shares)
+			if err != nil {
+				t.Error(err)
+			}
+
+			// check that the namesapces and data are the same
```

Comment:
[nit] typo
```suggestion
			// check that the namespaces and data are the same
```

something must be misconfigured with our misspell linter: https://github.com/celestiaorg/celestia-app/blob/721e31e2b9409ac5df10ad6ff7730c29c296a67c/.golangci.yml#L9

because I would expect these to be caught by that